# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/tests/wasm/jsapi/function/table.tentative.any.js
# Commit: 3b62bd57747f
# Full Hash: 3b62bd57747f21c800a93da325d999fa5efb3850
# Author: Jessica Tallon <jtallon@igalia.com>
# Date: 2021-09-03 03:51:58
# Regressor Bug: 1715275
# File Overlap Count: 1
# Description:
#   Bug 1715275 - Add support for a JS WebAssembly.Function object r=rhunt
#   
#   This introduces a new Constructor WebAssembly.Function which is part of
#   the WASM JS-API type reflection work. This allows normal JS functions to
#   be wrapped with a type and used with WebAssembly objects (e.g. Tables).
# ==============================================================================

diff -r f6f4514d09bf -r 3b62bd57747f testing/web-platform/tests/wasm/jsapi/function/table.tentative.any.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/testing/web-platform/tests/wasm/jsapi/function/table.tentative.any.js	Thu Sep 02 15:58:18 2021 +0000
@@ -0,0 +1,30 @@
+// META: global=window,dedicatedworker,jsshell
+// META: script=/wasm/jsapi/assertions.js
+
+function testfunc(n) {}
+
+test(() => {
+    var table = new WebAssembly.Table({element: "anyfunc", initial: 3})
+    var func1 = new WebAssembly.Function({parameters: ["i32"], results: []}, testfunc)
+    table.set(0, func1)
+    var func2 = new WebAssembly.Function({parameters: ["f32"], results: []}, testfunc)
+    table.set(1, func2)
+    var func3 = new WebAssembly.Function({parameters: ["i64"], results: []}, testfunc)
+    table.set(2, func3)
+
+    var first = table.get(0)
+    assert_true(first instanceof WebAssembly.Function)
+    assert_equals(first, func1)
+    assert_equals(first.type().parameters[0], func1.type().parameters[0])
+
+    var second = table.get(1)
+    assert_true(second instanceof WebAssembly.Function)
+    assert_equals(second, func2)
+    assert_equals(second.type().parameters[0], func2.type().parameters[0])
+
+    var third = table.get(2)
+    assert_true(third instanceof WebAssembly.Function)
+    assert_equals(third, func3)
+    assert_equals(third.type().parameters[0], func3.type().parameters[0])
+
+}, "Test insertion into table")